astropy/units/tests/test_quantity_helpers.py

```diff
@@ -31,3 +31,10 @@ def test_allclose_isclose():
     c = [90, 200] * u.cm
     assert not u.allclose(a, c)
     assert not np.all(u.isclose(a, c))
+
+
+def test_full():
+    a = u.Quantity(1, u.m)
+    b = np.full(3, a, like=a)
+    assert b.unit == a.unit
+    assert len(b) == 3
```

Comment:
This file is really for `is_close`, etc., i.e., helper routines to deal with quantities rather than helper routines to deal with numpy functions (naming is hard!!). It should go in `test_quantity_non_ufuncs.py` (did I mention naming is hard?). 

Note that that file attempts to keep track of whether our coverage is complete, and the set that it things needs to be covered is calculated in `get_wrapped_functions` - you'd need to extend that to include the generation-like routines like `np.full` (also `np.zeros`, `np.arange`, etc.). It would make sense to create a new `TestCreatingFunctionsUsingLike` or so.